### PR TITLE
Event propagation fix for Dropdown component

### PR DESCRIPTION
**Pull request by sick-sad-world**
Fixed event propagation for onClick handler and provided correct target for option click handler

### CODE REVIEW

**Comment by cmugla** [comment]
I'm a bit confused what the bug with propagation was, can you provide a description of the bug you were seeing in the description?

Also, please add a test!

File: packages/core/src/components/dropdown/index.js [hunk 1]
```diff
@@ -308,7 +308,21 @@ class Dropdown {
...
 
-  update = () => {
+  update = options => {
+    if (options) {
+      this.settings = {
+        ...defaults,
+        ...options
+      };
+    }
+    markupTemplates.forEach(template => {
```
Comment (cmugla) [inline packages/core/src/components/dropdown/index.js]: is this copy/pasted from the constructor? If so, let's make it its own function.

File: packages/core/src/components/dropdown/index.js [hunk 2]
```diff
@@ -330,8 +344,6 @@ class Dropdown {
   };
 
   onClick = e => {
-    e.stopPropagation();
-    e.preventDefault();
```
Comment (cmugla) [inline packages/core/src/components/dropdown/index.js]: why are these removed?

**Comment by cmugla** [comment]
Just merged the fix for the deploy-preview - please pull master and push back up. Hopefully will do the trick

File: packages/core/src/components/dropdown/index.js (hunk 1 above)
Comment (sick-sad-world) [reply packages/core/src/components/dropdown/index.js]: done

File: packages/core/src/components/dropdown/index.js (hunk 2 above)
Comment (sick-sad-world) [reply packages/core/src/components/dropdown/index.js]: this is removed because click handler is listening entire body to handle "click outside" and this is removed that it would not interfere with other click listeners. For example form submit button e.t.c Found this issue while integrating Dropdown into Landlords form

File: packages/core/src/components/dropdown/index.js [hunk 3]
```diff
@@ -315,15 +317,9 @@ class Dropdown {
         ...options
       };
     }
-    markupTemplates.forEach(template => {
```
Comment (cmugla) [inline packages/core/src/components/dropdown/index.js]: 🤔 this was removed, is it no longer needed or should you be calling `this._renderMarkup`?

File: packages/core/src/components/dropdown/index.js (hunk 2 above)
Comment (cmugla) [reply packages/core/src/components/dropdown/index.js]: ah ok

**Comment by cmugla** [comment]
Also want to note here that the Error state looks weird:
- the error hint is nested within the dropdown
- the dropdown is cut off by the message
- the border color is not the same when open

Image 2019-12-04 at 3 53 54 PM

File: packages/core/src/components/dropdown/index.js (hunk 3 above)
Comment (sick-sad-world) [reply packages/core/src/components/dropdown/index.js]: nope. i re-tested this and figured out that if we wan't to apply new options we do not need to re-render anything. Just set the value and (renderSelected) will be applied and update the option list and (renderOption) also will be invoked

**Comment by sick-sad-world** [comment]
> Also want to note here that the Error state doesn't match the non-error states, is this intentional?:
> 
> * the error hint is nested within the dropdown
> * the dropdown is cut off by the message
> * the border color is not the same when open
> 
> Image 2019-12-04 at 3 53 54 PM

Nah, this is my bad - put incorrect markup. Hints in ray is adjacent to input control not inside. Will Fix this i docs asap
